plugins/tiddlywiki/markdown/markdown-it-katex.js

```diff
@@ -0,0 +1,266 @@
+/*\
+title: $:/plugins/tiddlywiki/markdown/markdown-it-katex.js
+type: application/javascript
+module-type: library
+
+Based on markdown-it-katex v2.0.0 by @waylonflinn https://github.com/waylonflinn/markdown-it-katex | MIT License
+\*/
+(function(){
+/* Process inline math */
+/*
+Like markdown-it-simplemath, this is a stripped down, simplified version of:
+https://github.com/runarberg/markdown-it-math
+
+It differs in that it takes (a subset of) LaTeX as input and relies on KaTeX
+for rendering output.
+*/
+
+/*jslint node: true */
+'use strict';
+
+var katex = require("$:/plugins/tiddlywiki/katex/katex.min.js");
+
+// Test if potential opening or closing delimieter
+// Assumes that there is a "$" at state.src[pos]
+function isValidDelim(state, pos) {
+    var prevChar, nextChar,
```

Comment:
Our usual approach is to leave 3rd party files untouched, using the tiddlywiki.files mechanism to prefix or suffix any necessary glue code. Is that approach practical here?

Reply:
Yes, I purposefully left the spacing alone in case the code needs to be diffed. The code contains several modifications from the original, so I placed attributions at the top of the file and above `math_inline_bloc()`. Is there anything else I should do?